### PR TITLE
Add pre-commit configuration to get recognized as hook

### PR DESCRIPTION
**Pull request by iranzo**
Adds pre-commit configuration to get enabled as hook

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-   id: yaspeller
+    name: yaspeller
+    entry: yaspeller
+    language: node
+    files: .*\.(md|markdown)$
+    args: ["--find-repeat-words", "--ignore-digits", "--ignore-urls", "--ignore-uppercase"]
+    require_serial: false
+    additional_dependencies: []
```
Comment (asottile) [inline .pre-commit-hooks.yaml]:
```suggestion
```

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-   id: yaspeller
+    name: yaspeller
+    entry: yaspeller
+    language: node
+    files: .*\.(md|markdown)$
```
Comment (asottile) [inline .pre-commit-hooks.yaml]:
```suggestion
    types: [markdown]
```

File: README.md [hunk 3]
```diff
@@ -24,6 +24,19 @@ Used API [Yandex.Speller](https://tech.yandex.ru/speller/doc/dg/concepts/About-d
 ## Using CLI
 `yaspeller [options] <file-or-directory-or-link...>`
 
+## Using with pre-commit:
+
+Add this to your `.pre-commit-config.yaml`:
+
+```yaml
+- repo: https://github.com/hcodes/yaspeller.git
+  rev: master # Use the sha / tag you want to point at or master
```
Comment (asottile) [inline README.md]: the default configuration should not suggest master -- [this has more details](https://pre-commit.com/#using-the-latest-version-for-a-repository) -- generally I usually put either some fake value that's invalid here (such as `''`) or the latest tag

File: .pre-commit-hooks.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    args: ["--find-repeat-words", "--ignore-digits", "--ignore-urls", "--ignore-uppercase"]
```
Comment (asottile) [inline .pre-commit-hooks.yaml]: should this be blank? just let the default options work?

File: .pre-commit-hooks.yaml (hunk 2 above)
Comment (asottile) [reply .pre-commit-hooks.yaml]: does this apply to non-markdown too? should this be `types: [text]`

**Comment by iranzo** [comment]
@asottile pushed new commit with your suggestions, thanks!

File: README.md [hunk 5]
```diff
@@ -30,10 +30,10 @@ Add this to your `.pre-commit-config.yaml`:
 
 ```yaml
 - repo: https://github.com/hcodes/yaspeller.git
-  rev: master # Use the sha / tag you want to point at or master
+  rev: '' # Use the sha / tag you want to point at
   hooks:
     - id: yaspeller
-      args: ["--find-repeat-words", "--ignore-digits", "--ignore-urls"]
+      args: []
```
Comment (asottile) [inline README.md]:
```suggestion
```

File: .pre-commit-hooks.yaml [hunk 6]
```diff
@@ -2,5 +2,5 @@
     name: yaspeller
     entry: yaspeller
     language: node
-    types: [markdown]
-    args: ["--find-repeat-words", "--ignore-digits", "--ignore-urls", "--ignore-uppercase"]
+    types: [text]
+    args: []
```
Comment (asottile) [inline .pre-commit-hooks.yaml]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -24,6 +24,18 @@ Used API [Yandex.Speller](https://tech.yandex.ru/speller/doc/dg/concepts/About-d
 ## Using CLI
 `yaspeller [options] <file-or-directory-or-link...>`
 
+## Using with pre-commit:
```
Comment (asottile) [inline README.md]: I would remove the `:` here -- you probably also want to link to pre-commit.com so it's more clear what this is referring to (and so others can find the appropriate documentation)

Comment (iranzo) [reply README.md]: Done, thanks!

**Comment by iranzo** [comment]
@asottile if you think it's ok like it's now, I'll remove the WIP from title

File: README.md [hunk 8]
```diff
@@ -24,7 +24,7 @@ Used API [Yandex.Speller](https://tech.yandex.ru/speller/doc/dg/concepts/About-d
 ## Using CLI
 `yaspeller [options] <file-or-directory-or-link...>`
 
-## Using with pre-commit:
+## Using with [pre-commit](http://pre-commit.com/)
```
Comment (asottile) [inline README.md]:
```suggestion
## Using with [pre-commit](https://pre-commit.com/)
```

**Comment by iranzo** [comment]
@hcodes does it looks good for you?

**Comment by hcodes** [comment]
👍

**Pull request merged**